File: summaryRanges.ts
function summaryRanges(nums: number[]): string[] {
    let collect: string = '';
    for(let i = 0; i < nums.length;  i++){
        if(nums[i]+1 === nums[i+1]){
            if(nums[i]-1 !== nums[i-1])
                collect += nums[i] + "->"
         }
        else{
                collect += nums[i] + " " 
        }
    }
    const arr: string[] = collect.split(" ")
    arr.pop()
    return arr
};

File: summaryRanges.py
class Solution(object):
    def summaryRanges(self, nums):
        st = ''
        if len(nums) < 1: 
            return nums

        for x in range(len(nums)-1):
            if nums[x]+1 == nums[x+1]:
                if nums[x]-1 != nums[x-1]:
                    st += str(nums[x]) + "->"
            else:
                st += str(nums[x]) + " "
                
        st += str(nums[len(nums)-1])
      
        return st.split(" ")